fix Lantids gaia and pt scoring

rzulian opened the pull request:
Fixed Lantids scoring on gaia and planet types. Fixed also data passed to viewer.

coyotte508 commented on src/player.ts:
```diff
@@ -159,6 +159,11 @@ export default class Player extends EventEmitter {
   }
 
+  get ownedPlanets(): GaiaHex[] {
+    return this.data.occupied.filter(hex => hex.data.planet !== Planet.Empty && hex.colonizedBy(this.player)
```
Maybe `this.data.occupied.filter(hex => hex.data.planet !== Planet.Empty && hex.isMainOccupier(this.player))`

And add `isMainOccupier` to gaia hex. It can just return `this.colonizedBy(player) && this.additionalMine !== player`. No need to test against specific factions

coyotte508 commented on src/player.ts:
```diff
@@ -422,7 +427,7 @@ export default class Player extends EventEmitter {
     // this is for roundboosters, techtiles and adv tile
     for (const event of this.events[Operator.Trigger]) {
-      if (Condition.matchesBuilding(event.condition, building, planet)) {
+      if (Condition.matchesBuilding(event.condition, building, planet, this.faction)) {
```
can be reverted

rzulian commented on src/enums.ts:
```diff
@@ -93,12 +93,12 @@ export enum Condition {
     }
     switch (condition) {
-      case Condition.MineOnGaia: return building === Building.Mine && planet === Planet.Gaia;
+      case Condition.MineOnGaia: return building === Building.Mine && planet === Planet.Gaia && faction !== Faction.Lantids;
```
mines on other's planets are not getting any effect for gaia and pt. But you are right, this is impacting also regular mines. Need to correct this

coyotte508 replied:
> it does not count for any effects that relate to how many planet types or Gaia planets you have colonized.

From what I understand, it does not count for how many planet types or how many gaia planets you have colonized. But it does count for when you build a new mine on a gaia planet, since it's not a "count" effect but a trigger effect.

rzulian commented:
For clarity and also to adhere to the boardgame, I've changed the resource to `mg` when referring to `mine on gaia`.
I've disabled the triggered Income form building when from an additionalMine.
I've simplified with your suggestions the `player.ownedPlanets`, which is used in `pt` and `g` count.

coyotte508 commented:
So, the only thing left is mg -> g.

I already specifically changed it in the past. And mg messes up with the tech tiles in the UI:

![image](https://user-images.githubusercontent.com/342922/43355775-bbbbfed2-9262-11e8-88b3-a8eda07a72b4.png)

rzulian commented:
Works just fine with font 11px.
On the opportunity to have a different symbol I strongly suggest we don't confuse people with the same `g` for different meanings. Again, even the boardgame have two different graphics. It costs nothing and add clarity, for the user and for the code.
<img width="210" alt="screen shot 2018-07-28 at 13 29 32" src="https://user-images.githubusercontent.com/29941924/43356107-f504b622-926a-11e8-8c8d-7b64ca95e261.png">

coyotte508 commented on src/player.ts:
```diff
@@ -419,10 +426,11 @@ export default class Player extends EventEmitter {
     for (const event of this.events[Operator.Trigger]) {
-      if (Condition.matchesBuilding(event.condition, building, planet)) {
+      // only new mine trigger event for Lantids in other's planet
+      if (Condition.matchesBuilding(event.condition, building, planet) && (!isAdditionalMine || (isAdditionalMine && event.condition === Condition.Mine))) {
```
`&& (!isAdditionalMine || (isAdditionalMine && event.condition === Condition.Mine))` can be replaced by `&& (!isAdditionalMine || event.condition === Condition.Mine)`
